superset/models/core.py

```diff
@@ -457,28 +467,28 @@ def url(self):
                         )
                 except Exception:
                     pass
-        return "/superset/dashboard/{}/".format(self.slug or self.id)
+        return f"/superset/dashboard/{self.slug or self.id}/"
 
     @property
-    def datasources(self):
+    def datasources(self) -> Set[Optional["BaseDatasource"]]:
         return {slc.datasource for slc in self.slices}
 
     @property
-    def charts(self):
+    def charts(self) -> List[Optional["BaseDatasource"]]:
         return [slc.chart for slc in self.slices]
 
     @property
-    def sqla_metadata(self):
+    def sqla_metadata(self) -> None:
         # pylint: disable=no-member
         metadata = MetaData(bind=self.get_sqla_engine())
         return metadata.reflect()
```

Comment:
if this returns none, why do we even have a `return metadata.reflect()` here? Can we just remove the return?

Reply:
I agree, just remove `return` as it's not really a "returner" method.